lib/rspec/core/formatters/exception_presenter.rb

```diff
@@ -31,8 +31,31 @@ def colorized_message_lines(colorizer=::RSpec::Core::Formatters::ConsoleCodes)
           end
         end
 
-        def formatted_backtrace
-          backtrace_formatter.format_backtrace(exception_backtrace, example.metadata)
+        def formatted_backtrace(exception = @exception)
+          lines = backtrace_formatter.format_backtrace((exception.backtrace || []), example.metadata)
+          potential_lines = []
+
+          while exception.respond_to?(:cause) && exception.cause
+            exception = exception.cause
+            cause = []
+            cause << '------------------'
+            cause << '--- Caused by: ---'
+            cause << "#{exception_class_name(exception)}:" unless exception_class_name(exception) =~ /RSpec/
+
+            encoded_string(exception.message.to_s).split("\n").each do |line|
+              cause << "  #{line}"
+            end
+
+            cause << ("  " + backtrace_formatter.format_backtrace(exception.backtrace, example.metadata).first)
+
+            potential_lines << cause
+          end
+
+          unless potential_lines.empty?
+            lines += potential_lines.last
+          end
```

Comment:
If you're going to only take the last cause, why bother generating the string for all of them? It would seem better to loop over the causes until you've found the last, then do the string generation and add the last line directly to `lines`.